### ISSUE TITLE
[Feature Request]: Automation in automatic addition of databases

### ISSUE THREAD

**Issue by YoungBarick** [RedisInsight/RedisInsight#3452]
Hello! I'm facing a problem with automatically adding a connection to redis. I use curl request, which contains: user, password, host and ca certificate. The problem is that with such a request I get a 500 error. However, this error does not exist if I manually confirm “I have read and understood the Terms.” Can you tell me if there is an api endpoint so that I can automatically confirm “I have read and understood the Terms” using a curl request? Or maybe somebody can send me some documentation with API endpoints in Redisinsight?

**Comment by J5Dev** [comment]
Hi yes you can, you simply need to do something like the following, adjusting the values for your liking.

```
curl -XPATCH 'http://<yourendpoint>/api/settings' \
  -H 'Content-Type: application/json' \
  --data-raw '{"agreements":{"eula":true,"analytics":false,"notifications":false,"encryption":false}}'
  ```
  
  Replace your endpoint with the host:port, so if running local you would replace it with localhost:5540, and of course adjust it to https if running it secured.

**Comment by eerhardt** [comment]
I'm hitting this same problem. I want to seed the Redis Insight instance with a Redis connection, but the connection contains a password.

Request:
```json
[{"host":"redis","port":6379,"name":"redis","db":0,"username":null,"password":"XXX","connectionType":"STANDALONE"}]
```

Response:
```json
 {"total":1,"success":[],"partial":[],"fail":[{"index":0,"status":"fail","host":"redis","port":6379,"errors":[{"message":"Unsupported encryption strategy","name":"UnsupportedEncryptionStrategy","statusCode":500}]}]}
```

Is it possible to add a database with a password, but without sending a request to `/api/settings` first?

My guess is the exception is happening from this code:

https://github.com/RedisInsight/RedisInsight/blob/b551c6adaba198bdd7488116bb29e6998c489a58/redisinsight/api/src/modules/encryption/encryption.service.ts#L47-L61

cc @ArtemHoruzhenko as the author

**Comment by ArtemHoruzhenko** [comment]
@eerhardt unfortunately it is not possible at the moment. you have to provide consent first
Anyway  `http://<yourendpoint>/api/settings` is idempotent endpoint so you can add it to your initial script before database creation

**Comment by eerhardt** [comment]
Thanks for the info @ArtemHoruzhenko. We have a Redis and Redis Insight integration in .NET Aspire, which allows developers to easily and quickly spin up Redis and Redis Insight containers locally. It automates connecting the Redis Insight instance to a Redis database, so I added a call to `/api/settings` to ensure the settings are initialized before sending the `/api/databases/import` request.

Just one note, since .NET Aspire is doing this on behalf of a developer, it sends `eula: false`, since the tool isn't accepting the EULA on behalf of the developer that is using .NET Aspire. This seems to work. I just wanted to check that this is OK to do.

**Comment by ArtemHoruzhenko** [comment]
@eerhardt I don't think so. Not sure that users will be prompted with eula in this case

**Comment by eerhardt** [comment]
> Not sure that users will be prompted with eula in this case

From my testing, they aren't prompted.

**Comment by ArtemHoruzhenko** [comment]
@eerhardt I would suggest you to wait until we complete this feature request (I think it will be soon). With new functionality you will be able to create databases via environment variables.

**Comment by eerhardt** [comment]
I opened an issue for [[Bug]: Posting eula: false to /api/settings suppresses the first run dialog (RedisInsight/RedisInsight#4340)](https://github.com/RedisInsight/RedisInsight/issues/4340).

We will update .NET Aspire to use the new environment variable feature when it comes. It will greatly reduce our complexity as well, so that is a welcome improvement!

For now, I am going to workaround this issue by sending an `/api/settings` request to initialize encryption, so we can create a redis connection using a password.

**Comment by ViktarStarastsenka** [comment]
Redis Insight 2.68 introduces a way to preconfigure database connections. More information is described here - https://redis.io/docs/latest/operate/redisinsight/configuration/#:~:text=true-,Preconfigure%20database%20connections,-Redis%20Insight%20allows.
